### ISSUE TITLE
Terminal missing energy

### ISSUE THREAD

**Issue by totalschaden** [TooAngel/screeps#47]
Manual command to move Energy to a Terminal would be nice.
Automatic move Energy to a Terminal until X is reached (config) would be nice.

My Terminal is at 20692 Energy for a Week now and its not selling stuff because to low Energy.

**Comment by mschultheiss83** [comment]
Graz, mine at 
292,400 × e
6,600 × h
1,000 × L
doing nothing too and missing in storage for upgrader!

@TooAngel maybe you could add an equaliser between storage and terminal with an min and max thresholds, e.g. carry creeps in rooms with energy more than maximum energy threshold need to move energy to next room below minimum threshold?

**Comment by TooAngel** [comment]
There is some kind of logic in the `role_mineral`. Not activated by default, because of CPU issues.

That role is responsible for filling labs with minerals and energy, this could also do some equalizing.
If you like have a look and improve :-)

**Comment by mschultheiss83** [comment]
oh my god, i did looked to it but why did you define 11 functions inside of Creep.prototype.handleMineralCreep!
it will take some time to refactor Creep.prototype.handleMineralCreep

**Comment by TooAngel** [comment]
I guess every time when you have an 'oh my good' moment (there will be a couple more), my answer will be 'It is a grown system'.

In this case: Some weeks ago I migrated the roles from a `require('role_' + name)` to the roles dictionary. In the single modules I had a couple of methods defined in the global namespace. My main aim was to get the migration done, so I skipped cleaning up each role and went for ugly but quick and well working solutions. The mineral handling is one of the worst implemented parts.

Rewrite it, strip out what you don't like, don't understand and don't want anyone to see. It is not enabled by default and if you enable it, it will only be shit :-)
